### PR TITLE
*: turn on tests and use testify

### PR DESCRIPTION
**Pull request by tisonkun**

### CODE REVIEW

File: terror_test/terror_test.go [hunk 1]
```diff
@@ -15,35 +15,30 @@ package terror_test
 
 import (
 	"encoding/json"
+	"github.com/stretchr/testify/suite"
```
Comment (disksing) [inline terror_test/terror_test.go:18]: 💯

File: terror_test/terror_test.go [hunk 2]
```diff
@@ -81,85 +76,96 @@ func (s *testTErrorSuite) TestTraceAndLocation(c *C) {
...
+	errWithCause := errors.Annotate(err, cause.Error())
+	s.NotNil(errWithCause)
+
+	s.Equal("load from etcd meet error: [member:ErrGetLeader]fail to get leader", errWithCause.Error())
 }
+
+
+func TestExampleTestSuite(t *testing.T) {
+	suite.Run(t, new(TErrorTestSuite))
+}
```
Comment (disksing) [inline terror_test/terror_test.go:171]: no new line at end of file?

Comment (tisonkun) [reply terror_test/terror_test.go:171]: I can attach one. `go fmt` doesn't complain it.

**Comment by tisonkun** [comment]
cc @lysu @kennytm

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-  - 1.5.x
-  - 1.6.x
-  - 1.7.x
-  - 1.8.x
-  - 1.9.x
-  - 1.10.x
-  - tip
+  - 1.14.x
+  - 1.15.x
+  - 1.16.x
```
Comment (kennytm) [inline .travis.yml:9]: i think we should just test on `1.13.x` (used to build TiDB currently) and the latest stable version.

File: terror_test/terror_test.go [hunk 4]
```diff
@@ -81,85 +76,96 @@ func (s *testTErrorSuite) TestTraceAndLocation(c *C) {
...
 	_, f, l, _ := runtime.Caller(0)
 	terr, ok := errors.Cause(err).(*errors.Error)
-	c.Assert(ok, IsTrue)
+	s.True(ok)
+
 	file, line := terr.Location()
-	c.Assert(file, Equals, f)
-	c.Assert(line, Equals, l-1)
+	s.Equal(f, file)
+	s.Equal(l -1 , line)
```
Comment (kennytm) [inline terror_test/terror_test.go:146]:
```suggestion
	s.Equal(l - 1, line)
```

File: .travis.yml (hunk 3 above)
Comment (tisonkun) [reply .travis.yml:9]: how can I specify the latest stable version? `1.16.x` or `tip` or others? I think tip is the latest devel version and cause quite a lot of time to prepare the env.

Comment (tisonkun) [reply .travis.yml:9]: or could you please comment with a multiline suggestion?

Comment (kennytm) [reply .travis.yml:9]: @tisonkun "stable" or "1.x". https://github.com/travis-ci/gimme#asking-gimme-what-a-version-is

Comment (tisonkun) [reply .travis.yml:9]:
```suggestion
  - 1.13.x
  - stable
```

Comment (tisonkun) [reply .travis.yml:9]:
```suggestion
  - 1.13.x
  - stable
```

**Comment by tisonkun** [comment]
@disksing @kennytm PTAL

**Review by disksing — APPROVED** [review]

**Pull request merged**